clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -3835,6 +3835,31 @@ public void testFetchCompletedBeforeHandlerAdded() {
         assertEquals(1, fetcher.sendFetches());
     }
 
+    @Test
+    public void testCorruptMessageError() {
+        buildFetcher();
+        assignFromUser(singleton(tp0));
+        subscriptions.seek(tp0, 0);
+
+        assertEquals(1, fetcher.sendFetches());
+        assertFalse(fetcher.hasCompletedFetches());
+
+        // Prepare a response with the CORRUPT_MESSAGE error.
+        client.prepareResponse(fullFetchResponse(
+                tp0,
+                buildRecords(1L, 1, 1),
+                Errors.CORRUPT_MESSAGE,
+                100L, 0));
+        consumerClient.poll(time.timer(0));
+        assertTrue(fetcher.hasCompletedFetches());
+
+        // Trigger the exception.
+        assertThrows(KafkaException.class, () -> {
+            Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> partitionRecords =
+                    fetchedRecords();
```

Comment:
nit: just calling fetchedRecords() should work here, or is compiler not liking it and forcing the return value assignment?

Reply:
Agree, made the change to discard the return value.